src/tribler/core/components/bandwidth_accounting/db/database.py

```diff
@@ -91,8 +93,12 @@ def get_latest_transaction(self, public_key_a: bytes, public_key_b: bytes) -> Ba
         :param public_key_b: The public key of the party receiving the bandwidth.
         :return The latest transaction between the two specified parties, or None if no such transaction exists.
         """
-        db_obj = self.BandwidthTransaction.get(public_key_a=public_key_a, public_key_b=public_key_b)
-        return BandwidthTransactionData.from_db(db_obj) if db_obj else None
+        db_txs = select(tx for tx in self.BandwidthTransaction
+                        if public_key_a == tx.public_key_a and public_key_b == tx.public_key_b) \
+            .order_by(desc(self.BandwidthTransaction.timestamp)) \
+            .limit(1)
+
+        return BandwidthTransactionData.from_db(db_txs[0]) if len(db_txs) == 1 else None
```

Comment:
**nitpick:** It is a bit more convenient to use `.first()` instead of `.limit(1)`, this way the next line will be a bit simpler:

```suggestion
            .first()

        return BandwidthTransactionData.from_db(db_tx) if db_tx else None
```

`.first` returns the first object (if exists) or None